### ISSUE TITLE
Warn about supertrait bound duplicated in function trait bound (`where Self: Supertrait`)

### ISSUE THREAD

**Issue by jendrikw** [rust-lang/rust-clippy#8229]
### Summary

This could be covered under `trait_duplication_in_bounds`.

### Lint Name

trait_duplication_in_bounds

### Reproducer

I tried this code: [play](https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&code=%23!%5Bwarn(clippy%3A%3Astyle%2C%20clippy%3A%3Apedantic)%5D%0Atrait%20T%3A%20Default%20%7B%0A%20%20%20%20fn%20f()%20where%20Self%3A%20Default%3B%0A%7D)

```rust
#![warn(clippy::style, clippy::pedantic)]
trait T: Default {
    fn f() where Self: Default;
}
```

I expected to see this happen:
`warning: this trait bound is already specified in the where clause`

Instead, this happened:
no warning.

### Version

```text
rustc 1.57.0 (f1edd0429 2021-11-29)
binary: rustc
commit-hash: f1edd0429582dd29cccacaf50fd134b05593bd9c
commit-date: 2021-11-29
host: x86_64-unknown-linux-gnu
release: 1.57.0
LLVM version: 13.0.0
```
